python/lsst/daf/butler/dimensions/_records.py

```diff
@@ -299,6 +288,26 @@ def __init__(self, **kwargs: Any):
                 tuple(kwargs[dimension] for dimension in self.definition.required.names),
             ),
         )
+        # Don't need the primary key value aliased to the dimension name
+        # anymore.
+        kwargs.pop(self.definition.name, None)
+
+        for name in self.__slots__:
+            # Note that we remove from kwargs as we go, to make sure there's
+            # nothing left at the end.
+            object.__setattr__(self, name, kwargs.pop(name, None))
+        if self.definition.temporal is not None and self.timespan is None and "datetime_begin" in kwargs:
+            object.__setattr__(
+                self,
+                "timespan",
+                Timespan(
+                    kwargs.pop("datetime_begin"),
+                    kwargs.pop("datetime_end"),
+                ),
+            )
```

Comment:
There's a small change in behavior here, not sure if it is intended to enforce that any Timespan here will always be bounded or not.

Previously if only `datetime_begin` was passed without also including `datetime_end`, it would create a Timespan without an upper bound.  Now it throws KeyError in this case instead.